### PR TITLE
Use .cozmars directory

### PR DESCRIPTION
**Pull request by anjok**
- should use a directory instead of a bunch of files
- fix typo in README.md
- remove comment from conf.json

### CODE REVIEW

**Comment by anjok** [comment]
Actually, please don't pull this yet. I can't use different branches for the different issues as I don't know your schedule and they almost all overlap.

So what I did here:

- added gettext + locale
- moved all .tmpl to .html, as they are easier to view in the finder preview
- fixed all Chinese text to google-translated gettext messages

The next step is to generate the locales/base.dot file and add the various translations.

Please don't pull this yet, as I can't test any of this due of a lack of a configured raspi. I still need some parts.

But if possible, don't create conflicts for me yet...

**Comment by anjok** [comment]
Unfortunately you would need to regenerate the Chinese text from the to-be-created gettext files. Since I have no idea what the characters are, I would just create a mess and it would take ages.

**Comment by anjok** [comment]
@hyansuper  ok, localisation seems to work now. It would be great if you could take a look.

**Comment by hyansuper** [comment]
thank you for the effort, I am looking at it.
I have not used gettext before, may take a while

**Comment by anjok** [comment]
Please re-fetch the PR, I squashed the commits and rebased on your latest code.

Note that it's *very* hard for me to merge incoming changes of yours into my branch as I have no idea what your code actually spells... so please decide soon if you want to do the localisation or not. If not, I have to keep my own fork and won't be able to contribute.

I was working with this here:

https://phrase.com/blog/posts/translate-python-gnu-gettext/

it's basically:

`
cd rcute_cozmars_server
(wherever this is)/pygettext.py -d base -o locales/base.pot *.py *.html
`

Then generate loc files for each non-english language and fill in the blanks. It's really easy to use.

You'd need to proofread the English (as they are the keys, they need to be pretty good) and fill in the generated Chinese.

File: rcute_cozmars_server/static/servo.html [hunk 1]
```diff
@@ -1,65 +1,76 @@
...
 			return;
-		}
+		}}
 		stub = new RPCClient(ws);
-	})();
+	}})();
 </script>
-<h1>舵机校准</h1>
-<p><!-- 出厂时舵机已经调试好，一般不需要再进行校准<br> -->只有在 <a href='test'>测试</a> 中舵机无法到达指定位置时才需要校准</p><p style='background-color:yellow;text-align:left;color:red;padding:10px;'>警告: 点击保存后将无法恢复原设置！</p>
-<h2>头部&nbsp;&nbsp;&nbsp;&nbsp;<span id='head_val'>0°</span></h2>
```
Comment (hyansuper) [inline rcute_cozmars_server/static/servo.html:55]: did you missd line 55~56, they are for head servo calibration

File: rcute_cozmars_server/util.py [hunk 2]
```diff
@@ -1,5 +1,27 @@
...
+except Exception as e:
+    _ = gettext.gettext
+    _gettext = gettext.gettext
+
+def replace_gettext(match):
+    match = match.group(2)
+    return _gettext(match)
+
+def parsed_template(name, **kwargs):
+    with open(util.static("{}.html".format(name))) as file:
```
Comment (hyansuper) [inline rcute_cozmars_server/util.py:20]: "util.static" -> just "static"

File: rcute_cozmars_server/util.py [hunk 3]
```diff
@@ -1,5 +1,27 @@
 from os import path
 from PIL import Image, ImageFont, ImageDraw
+import gettext
+import locale
+
```
Comment (hyansuper) [inline rcute_cozmars_server/util.py:5]: import re

File: rcute_cozmars_server/static/test.html [hunk 4]
```diff
@@ -1,30 +1,54 @@
...
+	<tr><td>{_("Button")}</td><td id='button'></td></tr>
+	<tr><td>{_("Distance")}</td><td id='sonar'></td></tr>
+	<tr><td>{_("Infrared")}</td><td id='ir'></td></tr>
+</table>
+<h3>{_("Servo test")}</h3>
+	<button onclick='robot.head.angle(30)'>{_("Head up")}</button>
+	<button onclick='robot.head.angle(-30)'>{_("Head down")}</button>
+	<button onclick='robot.lift.height(1)'>{_("Lift up")}</button>
+	<button onclick='robot.lift.height(0)'>{_("Lift down")}</button>
+<h3>{_("Motor test")}  <input id='sec' type='number' size='1' value='2'/>{_("Speed")}</h3>
```
Comment (hyansuper) [inline rcute_cozmars_server/static/test.html:30]: "Speed" -> "second"

File: requirements.txt [hunk 5]
```diff
@@ -4,6 +4,8 @@ gpiozero
 picamera
 sounddevice
 sanic
+gettext
```
Comment (hyansuper) [inline requirements.txt:7]: no need to require gettext and locale, they are builtin with python3, they can't be installed individually

File: rcute_cozmars_server/__main__.py [hunk 6]
```diff
@@ -1,12 +1,14 @@
 import asyncio
 import sanic
-
+import re
```
Comment (hyansuper) [inline rcute_cozmars_server/__main__.py:3]: move this line to file util.py

File: rcute_cozmars_server/static/index.html [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+					<form id='upgrade' action=' upgrade' style="display:none">
+						<input type='submit' value="{_("Update")}" />
+					</form>
+				</td>
+			</tr>
+		</table>
+		<h2>{_("Settings")}</h2>
+		<a href='wifi'><button>{_("WiFi Settings")}</button></a>
+		<a href='test'><button>{_("Functional Test")}</button></a>
+		<a href='servo'> <button{_("Servo calibration")}</button></a><a href='motor'><button>{_("Motor calibration")}</button></a>
```
Comment (hyansuper) [inline rcute_cozmars_server/static/index.html:42]: "<button" -> left a ">" sign

File: rcute_cozmars_server/static/motor.html [hunk 8]
```diff
@@ -1,66 +1,85 @@
...
+	js_alter(typeof msgpack=='undefined','static/msgpack.min.js');
+	js_alter(typeof RPCClient=='undefined','static/client.js');
+	// js_alter(typeof Cozmars=='undefined','static/cozmars.js');
+	</script>
+	<script defer>
 	var stub, conf, speed=[0,0];
-	function reset_config() {
-		return fetch('http://'+location.host+'/conf').then(j=>j.json()).then(c=>{
+	function reset_config() {{
+		return fetch('http://'+location.host+'/config').then(j=>j.json()).then(c=>{{
```
Comment (hyansuper) [inline rcute_cozmars_server/static/motor.html:27]: config -> conf

File: rcute_cozmars_server/static/servo.html [hunk 9]
```diff
@@ -1,65 +1,76 @@
...
+	function js_alter(b, src){{if(b)document.write(unescape("%3Cscript src='"+src+"' type='text/javascript'%3E%3C/script%3E"));}}function q(d){{return document.querySelector(d)}}function sleep(sec){{return new Promise((r,j)=>{{setTimeout(r,sec*1000)}})}}
 	js_alter(typeof msgpack=='undefined', 'static/msgpack.min.js');
 	js_alter(typeof RPCClient=='undefined', 'static/client.js');
 	// js_alter(typeof Cozmars=='undefined', 'static/cozmars.js');
 </script><script defer>
 	var stub, conf;
-	function reset_config() {
-		return fetch('http://'+location.host+'/conf').then(j=>j.json()).then(c=>{
+	function reset_config() {{
+		return fetch('http://'+location.host+'/config').then(j=>j.json()).then(c=>{{
```
Comment (hyansuper) [inline rcute_cozmars_server/static/servo.html:24]: config -> conf

File: rcute_cozmars_server/static/test.html [hunk 10]
```diff
@@ -1,30 +1,54 @@
 <html><head><title>Cozmars</title></head><body>
+<style>button{{margin-left: 10px;margin-right: 10px;}}</style>
 <script src="https://cdn.jsdelivr.net/npm/@ygoe/msgpack@1.0.2/msgpack.min.js"></script>
 <script src="https://cdn.jsdelivr.net/gh/hyansuper/wsmprpc/js/client.min.js"></script>
-<script src="https://cdn.jsdelivr.net/gh/hyansuper/rcute-cozmars-server/rcute_cozmars_server/static/cozmars.min.js"></script>
+<script src="https://cdn.jsdelivr.net/gh/hyansuper/rcute-cozmars-server/server/static/cozmars.min.js"></script>
```
Comment (hyansuper) [inline rcute_cozmars_server/static/test.html:5]: this line should not change

File: rcute_cozmars_server/static/test.html (hunk 4 above)
Comment (hyansuper) [inline rcute_cozmars_server/static/test.html:30]: _("Speed") => _("seconds")

File: rcute_cozmars_server/util.py (hunk 3 above)
Comment (hyansuper) [inline rcute_cozmars_server/util.py:5]: import re

File: requirements.txt (hunk 5 above)
Comment (hyansuper) [inline requirements.txt:7]: gettext and locale is builtin with python3. don't need to require gettext and locale, they can't be installed individually

**Review by hyansuper — CHANGES_REQUESTED** [review]
just a few

**Comment by hyansuper** [comment]
sorry I made some duplicated change requests

**Comment by anjok** [comment]
ok, will fix and push later

**Comment by anjok** [comment]
@hyansuper please review again.

**Comment by anjok** [comment]
also, see:

https://stackoverflow.com/questions/27214065/how-to-docstring-in-python-for-multiple-languages

for the doctoring comments.

**Review by hyansuper — APPROVED** [review]

**Comment by hyansuper** [comment]
@anjok  did you have to hack pygettext.py in order to handle .html? 
I think this command also works `pygettext -d base -o locales/base.pot *.py static/*.html `
but a few english aren't extracted in html, especially those in javascript, maybe it has to do with "quote in quote in quoat...". I have to manually type into .pot file

**Comment by anjok** [comment]
Oh ok. I didn't verify it got everything. Worst case one need to put the error strings in a server side duct

**Pull request merged**